include/bm/bm_sim/headers.h

```diff
@@ -280,6 +282,35 @@ class Header : public NamedP4Object {
   //! string. The name is of the form <hdr_name>.<f_name>.
   const std::string get_field_full_name(int field_offset) const;
 
+  // The header and structure are represented by the class Header in bmv2.
+  // The difference between the header and the structure is in the metadata variable:
+  //  metadata = 0 for the header
+  //  metadata = 1 for the structure.
+  std::string show() const override {
+
+    if (!is_metadata() && !is_valid()) {
+      std::string name = get_name();
+      return "Header " + name.substr(0,name.rfind('_')) + " is not valid";
+    }
```

Comment:
I think the code should be more uniform and look like this:
 * if it's a header (i.e. `metadata` is false), print validity information first
 * then print all the *non-hidden* fields, so all fields up to the hidden offset

This special case is not needed IMO: even for a non-valid header, you should still honor the P4 program and print all field values